File: calc-sheet-column-number/calc-sheet-column-number.js
/**
 * @param {string} s
 * @return {number}
 */
function titleToNumber(s) {
    let result = 0;
    let unit = 1;

    for (let j = s.length - 1; j >= 0; j--){
        result += (convertCharToAscii(s.charAt(j)) - 64) * unit;
        unit *= 26;
    }

    return result;
}

function convertCharToAscii(inputChar){
    let index = 0;
    return inputChar.charCodeAt(index);
}

console.log(titleToNumber("A"));
console.log(titleToNumber("AB"));
console.log(titleToNumber("ZY"));
console.log(titleToNumber("AAA"));

File: calc-sheet-column-number/calc-sheet-column-number.dart
int titleToNumber(String s) {
  int result = 0;
  for (int i = 0; i < s.length; i++) {
    result = result * 26 + (s.codeUnitAt(i) - 64);
  }
  return result;
}

void main() {
  print(titleToNumber('A'));
  print(titleToNumber('AB'));
  print(titleToNumber('ZY'));
}
